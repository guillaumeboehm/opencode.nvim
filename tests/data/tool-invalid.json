[
  {
    "type": "server.connected",
    "properties": {}
  },
  {
    "type": "session.updated",
    "properties": {
      "info": {
        "title": "removal",
        "directory": "/Users/cam/Dev/neovim-dev/opencode.nvim",
        "projectID": "29d43526f88157cd4edd071b899dd01f240b771b",
        "version": "0.15.0",
        "id": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "time": {
          "created": 1760382404951,
          "updated": 1760382404951
        }
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "modelID": "claude-sonnet-4.5",
        "role": "assistant",
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "time": {
          "created": 1760386206864
        },
        "tokens": {
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0,
          "input": 0,
          "output": 0
        },
        "cost": 0,
        "path": {
          "cwd": "/Users/cam/Dev/neovim-dev/opencode.nvim",
          "root": "/Users/cam/Dev/neovim-dev/opencode.nvim"
        },
        "providerID": "github-copilot",
        "id": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "mode": "build"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "messageID": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "id": "prt_9df31dbba0016WMz330tCy15xM",
        "type": "step-start"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "callID": "toolu_01S6VFmzFVgdfCCYZDV3sokx",
        "tool": "edit",
        "state": {
          "status": "pending"
        },
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "messageID": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "id": "prt_9df31dc040016Voad2DzwhYBOm",
        "type": "tool"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "callID": "toolu_01S6VFmzFVgdfCCYZDV3sokx",
        "tool": "invalid",
        "state": {
          "status": "running",
          "input": {
            "tool": "edit",
            "error": "Invalid input for tool edit: JSON parsing failed: Text: {\"filePath\": \"/Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/streaming_renderer.lua\", \"newString\": \"---Event handler for permission.replied events\\n---Re-renders part after permission is resolved\\n---@param event table Event object\\nfunctio.\nError message: JSON Parse error: Unterminated string"
          },
          "time": {
            "start": 1760386212898
          }
        },
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "messageID": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "id": "prt_9df31dc040016Voad2DzwhYBOm",
        "type": "tool"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "callID": "toolu_01S6VFmzFVgdfCCYZDV3sokx",
        "tool": "invalid",
        "state": {
          "status": "completed",
          "title": "Invalid Tool",
          "input": {
            "tool": "edit",
            "error": "Invalid input for tool edit: JSON parsing failed: Text: {\"filePath\": \"/Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/streaming_renderer.lua\", \"newString\": \"---Event handler for permission.replied events\\n---Re-renders part after permission is resolved\\n---@param event table Event object\\nfunctio.\nError message: JSON Parse error: Unterminated string"
          },
          "output": "The arguments provided to the tool are invalid: Invalid input for tool edit: JSON parsing failed: Text: {\"filePath\": \"/Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/streaming_renderer.lua\", \"newString\": \"---Event handler for permission.replied events\\n---Re-renders part after permission is resolved\\n---@param event table Event object\\nfunctio.\nError message: JSON Parse error: Unterminated string",
          "metadata": {},
          "time": {
            "end": 1760386212900,
            "start": 1760386212898
          }
        },
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "messageID": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "id": "prt_9df31dc040016Voad2DzwhYBOm",
        "type": "tool"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "tokens": {
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0,
          "input": 0,
          "output": 0
        },
        "cost": 0,
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "messageID": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "id": "prt_9df31e425001bp5r5xQi4zr3Ph",
        "type": "step-finish"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "modelID": "claude-sonnet-4.5",
        "role": "assistant",
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "time": {
          "created": 1760386206864
        },
        "tokens": {
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0,
          "input": 0,
          "output": 0
        },
        "cost": 0,
        "path": {
          "cwd": "/Users/cam/Dev/neovim-dev/opencode.nvim",
          "root": "/Users/cam/Dev/neovim-dev/opencode.nvim"
        },
        "providerID": "github-copilot",
        "id": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "mode": "build"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "modelID": "claude-sonnet-4.5",
        "role": "assistant",
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "time": {
          "created": 1760386206864,
          "completed": 1760386212950
        },
        "tokens": {
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0,
          "input": 0,
          "output": 0
        },
        "cost": 0,
        "path": {
          "cwd": "/Users/cam/Dev/neovim-dev/opencode.nvim",
          "root": "/Users/cam/Dev/neovim-dev/opencode.nvim"
        },
        "providerID": "github-copilot",
        "id": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "mode": "build"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "modelID": "claude-sonnet-4.5",
        "role": "assistant",
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "time": {
          "created": 1760386206864,
          "completed": 1760386212951
        },
        "tokens": {
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0,
          "input": 0,
          "output": 0
        },
        "cost": 0,
        "path": {
          "cwd": "/Users/cam/Dev/neovim-dev/opencode.nvim",
          "root": "/Users/cam/Dev/neovim-dev/opencode.nvim"
        },
        "providerID": "github-copilot",
        "id": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "mode": "build"
      }
    }
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "modelID": "claude-sonnet-4.5",
        "role": "assistant",
        "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt",
        "time": {
          "created": 1760386206864,
          "completed": 1760386212951
        },
        "tokens": {
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0,
          "input": 0,
          "output": 0
        },
        "cost": 0,
        "path": {
          "cwd": "/Users/cam/Dev/neovim-dev/opencode.nvim",
          "root": "/Users/cam/Dev/neovim-dev/opencode.nvim"
        },
        "providerID": "github-copilot",
        "id": "msg_9df31cc90001HGn2UbFUgqJnLr",
        "mode": "build"
      }
    }
  },
  {
    "type": "session.idle",
    "properties": {
      "sessionID": "ses_6210836a9ffejgCtBDIpNCsYMt"
    }
  }
]
