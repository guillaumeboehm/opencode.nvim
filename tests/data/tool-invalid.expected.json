{"extmarks":[[1,2,0,{"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["BUILD","OpencodeMessageRoleAssistant"],[" claude-sonnet-4.5","OpencodeHint"],[" (2025-10-13 20:10:06)","OpencodeHint"],[" [msg_9df31cc90001HGn2UbFUgqJnLr]","OpencodeHint"]],"virt_text_hide":false,"priority":10,"virt_text_repeat_linebreak":false,"virt_text_win_col":-3,"ns_id":3,"virt_text_pos":"win_col","right_gravity":true}],[2,4,0,{"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_hide":false,"priority":4096,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"ns_id":3,"virt_text_pos":"win_col","right_gravity":true}],[3,5,0,{"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_hide":false,"priority":4096,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"ns_id":3,"virt_text_pos":"win_col","right_gravity":true}],[4,6,0,{"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_hide":false,"priority":4096,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"ns_id":3,"virt_text_pos":"win_col","right_gravity":true}],[5,7,0,{"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_hide":false,"priority":4096,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"ns_id":3,"virt_text_pos":"win_col","right_gravity":true}],[6,8,0,{"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_hide":false,"priority":4096,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"ns_id":3,"virt_text_pos":"win_col","right_gravity":true}],[7,9,0,{"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_hide":false,"priority":4096,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"ns_id":3,"virt_text_pos":"win_col","right_gravity":true}]],"timestamp":1760658432,"lines":["","----","","","**  tool** `invalid`","","> [!ERROR]",">","> Invalid input for tool edit: JSON parsing failed: Text: {\"filePath\": \"/Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/streaming_renderer.lua\", \"newString\": \"---Event handler for permission.replied events\\n---Re-renders part after permission is resolved\\n---@param event table Event object\\nfunctio.","> Error message: JSON Parse error: Unterminated string",""],"actions":[]}